assets/js/phoenix_live_view/hooks.js

```diff
@@ -135,9 +141,9 @@ Hooks.InfiniteScroll = {
       }
       scrollBefore = scrollNow
     }
-    window.addEventListener("scroll", this.onScroll)
+    scrollContainer.addEventListener("scroll", this.onScroll)
```

Comment:
I think you can avoid the complexity of walking back up the tree by setting the event listener on the `document` and passing `{ capture: true }` as options. Even if the scroll target _isn't_ the window, you will still have access to the scrolled element via `event.target`

Reply:
btw I looked into this a little bit. The issue with that approach is that if there are multiple scrollable containers on the page, the handler would still trigger even for unrelated scrolls. This could be worked around by checking if `e.target.contains(this.el)`, but I don't think that this would be a check that we'd want to do on every call. The search for the container happens only when the hook is first mounted, so that's fine I'd say.